### DISCUSSION TITLE
WebSocket backend ?

### DISCUSSION THREAD

**Discussion by ProbablyClem** [SeaQL/sea-streamer#12] [Ideas]
Hey guys,
What do you think about implenting a backend for WebSocket ?
I think that would be cool since websockets are basically streams/sockets just like the other backends and websockets are really popular nowadays.
It doesn't look very hard from my naive look, and I could maybe give it a try if you find it interresting and I find the time.
Thank you guys, you rock

**Comment by tyt2y3** [comment]
Thank you.
Not sure if it matches your imagination. But what I had in mind for WebSocket specifically, is to talk to frontend clients, e.g. browsers.
So architecturally we might want to implement a standalone web socket server that can relay from any backends.
For peer-to-peer communication, the nuisance might be in peer discovery.
For lightweight-ness, I think Redis is a pretty good option for same host streaming or streaming within a small network group.

**Reply by ProbablyClem** [reply]
Hmm, personnaly I was just thinking about an abstraction of the client side, like the kafka backend does...
However, your idea of a a standalone web socket server makes sens to me.
This looks like two distinct projects tho, maybe two distinct backends ? 
sea-streamer-websocket-client and sea-streamer-websocket-server ?

**Reply by tyt2y3** [reply]
Do you currently have a web socket server implementation you want to talk to?
What's the application you are building?
